script/scaffold/templates/device_automation/integration/device_automation.py

```diff
@@ -0,0 +1,195 @@
+"""Provides device automations for NEW_NAME."""
+from typing import Callable, Optional, List
+import voluptuous as vol
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_DOMAIN,
+    CONF_TYPE,
+    CONF_PLATFORM,
+    CONF_DEVICE_ID,
+    CONF_ENTITY_ID,
+    STATE_ON,
+    STATE_OFF,
+    SERVICE_TURN_ON,
+    SERVICE_TURN_OFF,
+)
+from homeassistant.core import HomeAssistant, Context, CALLBACK_TYPE
+from homeassistant.helpers import condition, entity_registry
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+from homeassistant.helpers.config_validation import (
+    DEVICE_ACTION_SCHEMA,
+    DEVICE_CONDITION_SCHEMA,
+)
+from homeassistant.components.automation import state
+from homeassistant.components.device_automation import (  # noqa  # pylint: disable=unused-import
+    TRIGGER_SCHEMA,
+)
+from . import DOMAIN
+
+# Types of supported triggers, conditions and actions
+TRIGGER_TYPES = {"turned_on", "turned_off"}
+CONDITION_TYPES = {"is_on"}
+ACTION_TYPES = {"turn_on", "turn_off"}
+
+TRIGGER_SCHEMA = TRIGGER_SCHEMA.extend({vol.Required(CONF_TYPE): TRIGGER_TYPES})
+
+CONDITION_SCHEMA = DEVICE_CONDITION_SCHEMA.extend(
+    {vol.Required(CONF_TYPE): CONDITION_TYPES}
+)
+
+ACTION_SCHEMA = DEVICE_ACTION_SCHEMA.extend({vol.Required(CONF_TYPE): ACTION_TYPES})
+
+
+async def async_get_triggers(hass: HomeAssistant, device_id: str) -> List[dict]:
+    """List device triggers for NEW_NAME devices."""
+    registry = await entity_registry.async_get_registry(hass)
+    triggers = []
+
+    # Get all the integrations entities for this device
+    for entry in entity_registry.async_entries_for_device(registry, device_id):
+        if entry.domain != DOMAIN:
+            continue
+
+        # Add triggers for each entity that belongs to this integration
+        # TODO add your own triggers. It's ok to not have any triggers.
+        triggers.append(
```

Comment:
Should these blocks just be part of the comment above? Or is there value in assuming/treating everything works like an on/off type of device?

Reply:
It was mainly to get tests going and show people how it should look.

You think that we should comment it out to emphasize that it's an example ?

Reply:
If you’re not worried that people will leave it in their implementations then leave it.

Reply:
Let's leave it in and see what the PRs do.